MantidQt/CustomInterfaces/src/SANSRunWindow.cpp

```diff
@@ -1702,18 +1700,21 @@ void SANSRunWindow::setGeometryDetails() {
     return;
   }
 
-  const auto &dets = monitorWs->getSpectrum(monitorWsIndex).getDetectorIDs();
-  if (dets.empty())
+  const auto &monitorDetectorIDs =
+      monitorWs->getSpectrum(monitorWsIndex).getDetectorIDs();
+  if (monitorDetectorIDs.empty())
     return;
 
   double dist_mm(0.0);
   QString colour("black");
-  try {
-    Mantid::Geometry::IDetector_const_sptr detector =
-        instr->getDetector(*dets.begin());
 
-    double unit_conv(1000.);
-    dist_mm = detector->getDistance(*source) * unit_conv;
+  const auto &spectrumInfo = sampleWs->spectrumInfo();
+
+  try {
+    const auto &detector = spectrumInfo.detector(*monitorDetectorIDs.begin());
+    const double unit_conv(1000.);
+    const auto &source = sampleWs->getInstrument()->getSource();
+    dist_mm = detector.getDistance(*source) * unit_conv;
```

Comment:
I think that this is not correct.
You query `SpectrumInfo::detector` with detector ID. It should be queried with a workspace index.

You can get the detector via
```cpp
...
const auto& detectorInfo = sampleWs->detectorInfo();
const auto& detector = detectorInfo.detector(detectorInfo.indexOf(detId);
```
 where detId is a detector ID. Note that `DetectorInfo::detector` does also not take a detector ID, but a separate index into DetectorInfo. You can convert from detector ID to detector index with `DetectorInfo::indexOf`.

Just to be sure that my assessment is correct, @SimonHeybrock  could you have a look at this.